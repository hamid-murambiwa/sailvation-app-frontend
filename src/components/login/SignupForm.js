/* eslint-disable*/

import React, { useReducer, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { postNewUser } from '../../redux/Users/users';
 

function SignupForm() {

  const dispatch = useDispatch();
  const [details, setDetails] = useState(
    {name: '', email: ''}
  );

  const submitHandler = (e) => {
    e.preventDefault();
    if ( details.name !== '' && details.email !== '' ) {
      dispatch(postNewUser(details));
    }
  }

  return (
    <form onSubmit={submitHandler}>
      <div className="form-inner">
        <h2>Signup</h2>
        {/* ERROR! */}
        <div className="form-group">
          <label htmlFor="name">
            Name:
            <input type="text" name="name" id="name" onChange={(e) => setDetails({ ...details, name: e.target.value })} value={details.name} />
          </label>
        </div>

        <div className="form-group">
          <label htmlFor="email">
            Email:
            <input type="email" name="email" id="email" onChange={(e) => setDetails({ ...details, email: e.target.value })} value={details.email} />
          </label>
        </div>

        <input type="submit" value="Signup" />
      </div>
    </form>
  );
}

export default SignupForm;
