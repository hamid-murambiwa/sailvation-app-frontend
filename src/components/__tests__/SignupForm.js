import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom/cjs/react-router-dom.min';
import { Provider } from 'react-redux';
import store from '../__mocks__/configureStore';
import SignupForm from '../login/SignupForm';

describe('Signup page is working fine', () => {
  test('renders Signup', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <SignupForm />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
