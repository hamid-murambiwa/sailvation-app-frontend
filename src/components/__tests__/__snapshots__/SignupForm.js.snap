// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page is working fine renders Login 1`] = `
<div
  className="container mt-5"
>
  <div
    className="m-4"
  >
    <h2>
      Already Have an Account?
    </h2>
    <a
      className="btn btn-success mt-3"
      href="/login"
      onClick={[Function]}
    >
      You can login from here
    </a>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-inner m-4"
    >
      <h2>
        Signup
      </h2>
      <div
        className="form-group"
      >
        <label
          htmlFor="name"
        >
          Name:
          <input
            id="name"
            name="name"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        className="form-group mt-3"
      >
        <label
          htmlFor="email"
        >
          Email:
          <input
            id="email"
            name="email"
            onChange={[Function]}
            type="email"
            value=""
          />
        </label>
      </div>
      <input
        className="btn btn-success mt-3"
        disabled={false}
        type="submit"
        value="Signup"
      />
    </div>
  </form>
</div>
`;

exports[`Signup page is working fine renders Signup 1`] = `
<div
  className="container mt-5"
>
  <div
    className="m-4"
  >
    <h2>
      Already Have an Account?
    </h2>
    <a
      className="btn btn-success mt-3"
      href="/login"
      onClick={[Function]}
    >
      You can login from here
    </a>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-inner m-4"
    >
      <h2>
        Signup
      </h2>
      <div
        className="form-group"
      >
        <label
          htmlFor="name"
        >
          Name:
          <input
            id="name"
            name="name"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        className="form-group mt-3"
      >
        <label
          htmlFor="email"
        >
          Email:
          <input
            id="email"
            name="email"
            onChange={[Function]}
            type="email"
            value=""
          />
        </label>
      </div>
      <input
        className="btn btn-success mt-3"
        disabled={false}
        type="submit"
        value="Signup"
      />
    </div>
  </form>
</div>
`;
